app/src/main/java/io/apicurio/registry/ccompat/rest/impl/CompatibilityResourceImpl.java

```diff
@@ -0,0 +1,77 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.ccompat.rest.impl;
+
+import io.apicurio.registry.ccompat.dto.CompatibilityCheckResponse;
+import io.apicurio.registry.ccompat.dto.SchemaContent;
+import io.apicurio.registry.ccompat.rest.CompatibilityResource;
+import io.apicurio.registry.ccompat.store.RegistryStorageFacade;
+import io.apicurio.registry.content.ContentHandle;
+import io.apicurio.registry.metrics.ResponseErrorLivenessCheck;
+import io.apicurio.registry.metrics.ResponseTimeoutReadinessCheck;
+import io.apicurio.registry.metrics.RestMetricsApply;
+import io.apicurio.registry.rules.RuleViolationException;
+import io.apicurio.registry.rules.RulesService;
+import io.apicurio.registry.types.ArtifactType;
+import org.eclipse.microprofile.metrics.annotation.ConcurrentGauge;
+import org.eclipse.microprofile.metrics.annotation.Counted;
+import org.eclipse.microprofile.metrics.annotation.Timed;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.interceptor.Interceptors;
+
+import static io.apicurio.registry.metrics.MetricIDs.*;
+import static org.eclipse.microprofile.metrics.MetricUnits.MILLISECONDS;
+
+/**
+ * @author Ales Justin
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+
+@ApplicationScoped
+@Interceptors({ResponseErrorLivenessCheck.class, ResponseTimeoutReadinessCheck.class})
+@RestMetricsApply
+@Counted(name = REST_REQUEST_COUNT, description = REST_REQUEST_COUNT_DESC, tags = {"group=" + REST_GROUP_TAG, "metric=" + REST_REQUEST_COUNT})
+@ConcurrentGauge(name = REST_CONCURRENT_REQUEST_COUNT, description = REST_CONCURRENT_REQUEST_COUNT_DESC, tags = {"group=" + REST_GROUP_TAG, "metric=" + REST_CONCURRENT_REQUEST_COUNT})
+@Timed(name = REST_REQUEST_RESPONSE_TIME, description = REST_REQUEST_RESPONSE_TIME_DESC, tags = {"group=" + REST_GROUP_TAG, "metric=" + REST_REQUEST_RESPONSE_TIME}, unit = MILLISECONDS)
+public class CompatibilityResourceImpl extends AbstractResource implements CompatibilityResource {
+
+    @Inject
+    RulesService rules;
+
+    @Inject
+    RegistryStorageFacade storage;
+
+    @Override
+    public CompatibilityCheckResponse testCompatibilityBySubjectName(
+            String subject,
+            String versionString,
+            SchemaContent request) throws Exception {
+
+        CompatibilityCheckResponse result = storage.parseVersionString(subject, versionString, version -> {
+            // TODO are we safe to assume AVRO?

```

Comment:
Yeah, they now support different types as well.
But is this reflected in their AIP?